pkg/controlplane/reconcilers/endpointsadapter.go

```diff
@@ -80,12 +78,9 @@ func (adapter *EndpointsAdapter) Update(namespace string, endpoints *corev1.Endp
 }
 
 // EnsureEndpointSliceFromEndpoints accepts a namespace and Endpoints resource
-// and creates or updates a corresponding EndpointSlice if an endpointSliceClient
-// exists. An error will be returned if it fails to sync the EndpointSlice.
+// and creates or updates a corresponding EndpointSlice. An error will be returned
+// if it fails to sync the EndpointSlice.
 func (adapter *EndpointsAdapter) EnsureEndpointSliceFromEndpoints(namespace string, endpoints *corev1.Endpoints) error {
-	if adapter.endpointSliceClient == nil {
-		return nil
-	}
```

Comment:
hmm, I _really_ don't want to pull back any additional non-test changes (and the unit test is panicking on a nil pointer I suspect is related to this.

I would prefer to keep the non-test change limited to the single line in pkg/controlplane/reconcilers/lease.go and unit test it differently if needed with a new unit test rather than backporting a whole stack of changes

Reply:
thanks, I was thinking about that and I didn't like to pull this 7 commits too ... I think that is a great idea, let me change it and add a new unit test